tools/generator/src/DTO/XcodeScheme.swift

```diff
@@ -29,25 +29,36 @@ No actions were provided for the scheme "\(name)".
 }
 
 extension XcodeScheme {
+    enum EnableBuildForValueMode {
+        case merge
+        case setIfAble
+    }
+
     /// Create a new scheme applying any default actions based upon the current scheme.
     var withDefaults: XcodeScheme {
         get throws {
             var buildTargets = [BazelLabel: XcodeScheme.BuildTarget]()
 
             func enableBuildForValue(
                 _ label: BazelLabel,
-                _ keyPath: WritableKeyPath<XcodeScheme.BuildFor, XcodeScheme.BuildFor.Value>
+                _ keyPath: WritableKeyPath<XcodeScheme.BuildFor, XcodeScheme.BuildFor.Value>,
+                _ mode: EnableBuildForValueMode = .merge
             ) throws {
-                do {
-                    try buildTargets[label, default: .init(label: label, buildFor: .init())]
-                        .buildFor[keyPath: keyPath]
-                        .merge(with: .enabled)
-                } catch XcodeScheme.BuildFor.Value.ValueError.incompatibleMerge {
-                    throw UsageError(message: """
-The `build_for` value, "\(keyPath.stringValue)", for "\(label)" in the "\(name)" Xcode scheme was \
-disabled, but the target is referenced in the scheme's \(keyPath.actionType) action.
-""")
+                var buildTarget = buildTargets[label, default: .init(label: label, buildFor: .init())]
+                switch mode {
+                case .merge:
+                    do {
+                        try buildTarget.buildFor[keyPath: keyPath].merge(with: .enabled)
+                    } catch XcodeScheme.BuildFor.Value.ValueError.incompatibleMerge {
+                        throw UsageError(message: """
+    The `build_for` value, "\(keyPath.stringValue)", for "\(label)" in the "\(name)" Xcode scheme was \
```

Comment:
Can you de-indent this?

Reply:
It looks like it is at the right indent. The syntax for properties that throw requires an extra level of indent. I can switch it to a function if you would like.

Reply:
I mean the string having indent on the left.